src/edu/psu/sweng500/team8/play/ActionManager.java

```diff
@@ -36,12 +36,13 @@ public void doUndo(CellGrid currentCellGridFromBoard) {
 		 * revert last action on the sudokuActionQueue, and put that action into
 		 * sudokuActionQueueForUndo
 		 */
-		SudokuAction lastAction = sudokuActionQueue.pop();
 		
-		if (null != lastAction) {
-			CellGrid previousCellGrid = lastAction.getCellGrid();
-			lastAction.setCellGrid(new CellGrid(currentCellGridFromBoard));
-			sudokuActionQueueForUndo.add(lastAction);
+		if (null != sudokuActionQueue.peek()) {

```

Comment:
name changed from queue to stack.
not a bug. setting the size cause the stack to be non-empty. but filled with nulls.
thus classic peek/pop should be used instead of isEmpty